iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/entity/patent/PatentExporterJob.java

```diff
@@ -0,0 +1,517 @@
+                .setRelClass(ResultResultProtos.ResultResult.Relationship.RelName.isRelatedTo.name())
+                .setSource(documentIdToExport)
+                .setTarget(patentIdToExport)
+                .setChild(false)
+                .setResultResult(OAFREL_RESULTRESULT)
+                .build();
+    }
+
+    private static JavaPairRDD<Text, Text> entitiesToExport(JavaRDD<DocumentToPatentWithIdsToExport> documentToPatentsToExportWithIds,
+                                                            JavaPairRDD<CharSequence, Patent> patentsById,
+                                                            String entityActionSetId) {
+        return documentToPatentsToExportWithIds
+                .mapToPair(x -> new Tuple2<>(x.getDocumentToPatent().getPatentId(), x))
+                .leftOuterJoin(patentsById)
+                .values()
+                .map(x -> {
+                    Optional<Patent> patent = x._2();
+                    String patentIdToExport = x._1().getPatentIdToExport();
+                    return new Tuple2<>(patent, patentIdToExport);
+                })
+                .distinct()
+                .filter(x -> x._1().isPresent())
+                .map(x -> new Tuple2<>(x._1().get(), x._2()))
+                .map(x -> {
+                    Patent patent = x._1();
+                    String patentIdToExport = x._2();
+                    return buildEntityAction(patent, patentIdToExport, entityActionSetId);
+                })
+                .mapToPair(PatentExporterJob::actionToTuple);
+    }
+
+    private static AtomicAction buildEntityAction(Patent patent, String patentIdToExport, String entityActionSetId) {
+        OafProtos.Oaf oaf = buildEntityOaf(patent, patentIdToExport);
+        return actionFactory.createAtomicAction(
+                entityActionSetId,
+                StaticConfigurationProvider.AGENT_DEFAULT,
+                patentIdToExport,
+                OafDecoder.decode(oaf).getCFQ(),
+                InfoSpaceConstants.QUALIFIER_BODY_STRING,
+                oaf.toByteArray()
+        );
+    }
+
+    private static OafProtos.Oaf buildEntityOaf(Patent patent, String patentIdToExport) {
+        return OafProtos.Oaf.newBuilder()
+                .setKind(KindProtos.Kind.entity)
+                .setEntity(buildOafEntity(patent, patentIdToExport))
+                .setLastupdatetimestamp(System.currentTimeMillis())
+                .build();
+    }
+
+    private static OafProtos.OafEntity buildOafEntity(Patent patent, String patentIdToExport) {
+        String now = DateTimeUtils.format(LocalDateTime.now());
+        return OafProtos.OafEntity.newBuilder()
+                .setType(TypeProtos.Type.result)
+                .setId(patentIdToExport)
+                .addCollectedfrom(OAF_ENTITY_COLLECTEDFROM)
+                .addPid(buildOafEntityPid(patent))
+                .setDateofcollection(now)
+                .setDateoftransformation(now)
```

Comment:
Ok, I'm setting both dates to static value

Reply:
This is quite OK for now. But could I ask you to introduce just one more modification by adding one new `dateOfCollection` job parameter and defining `2019-06-12T09:56` value within `workflow.xml` definition? 
This way we could quite easily override those dates when e.g. Harry provides a new EPO dump. Otherwise we would need to recompile.

Reply:
Done, to match with other patent exporter wf params I've named the parameter `patent_file_date_of_collection`.